### PR TITLE
Fix inviting users with undisclosed profiles

### PR DESCRIPTION
**Pull request by weeman1337**
Notes: Inviting users with undisclosed profiles now works

Closes vector-im/element-web#17269

---
Here's what your changelog entry will look like:

## 🐛 Bug Fixes
 * Inviting users with undisclosed profiles now works ([\#8504](https://github.com/matrix-org/matrix-react-sdk/pull/8504)). Fixes vector-im/element-web#17269. Contributed by @weeman1337.

### CODE REVIEW

File: src/utils/MultiInviter.ts [hunk 1]
```diff
@@ -168,10 +168,18 @@ export default class MultiInviter {
...
-                const profile = await this.matrixClient.getProfileInfo(addr);
-                if (!profile) {
-                    // noinspection ExceptionCaughtLocallyJS
-                    throw new Error("User has no profile");
+                try {
+                    await this.matrixClient.getProfileInfo(addr);
+                } catch (err) {
+                    // convert the errors so that they can be handled correctly later on
+                    switch (err.errcode) {
+                        case 'M_FORBIDDEN':
```
Comment (kerryarchibald) [inline src/utils/MultiInviter.ts:176]: Might be nice to add a comment explaining why this maps to undisclosed

**Review by kerryarchibald — COMMENTED** [review]
Looks good! Can you please add some test cases for this?

File: src/utils/MultiInviter.ts (hunk 1 above)
Comment (weeman1337) [reply src/utils/MultiInviter.ts:177]: I hope the comment is better now.

**Review by gsouquet — APPROVED** [review]
🙌  Looks great to me!

**Pull request merged**